### DISCUSSION TITLE
I keep getting a 404 error and I don't understand what the issue is.

### DISCUSSION THREAD

**Discussion by jcooler** [vercel/community#6676] [Help] [answered]
### Summary

I'm trying to deploy my backend for a MERN app on vercel and the frontend has been uploaded fine, but that backend keeps giving me a 404 error after it is "deployed" the page just says:

404: NOT_FOUND
Code: NOT_FOUND
ID: iad1::sc2lt-1714420227245-7408f7bc722c

I don't know what this means or why it isn't working. I've watched several youtube videos of people doing this seamlessly and I don't understand why I can't do it.

### Steps to Reproduce

I'm not getting any error messages. When I upload my backend to vercel it takes it, and says it is deployed, but just keeps saying 404 not found. Is there a time period that this works or? I've been at this for over 8 hours today.

![image](https://github.com/vercel/community/assets/84881811/19a7a596-2104-4c19-8359-1db001f3d15e)

**Comment by ososnovsky** [comment]
add vercel.json file with the following:
{
  "rewrites": [
    {
      "source": "/(.*)",
      "destination": "/index.html"
    }
  ]
}

**Comment by amyegan** [comment, marked as answer]
Backend code would use one of our available [runtimes](https://vercel.com/docs/functions/runtimes#). In your case, it would be the [Node.js Runtime](https://vercel.com/docs/functions/runtimes/node-js) which will automatically use code in the `/api` directory as serverless functions. If your code is not in the `/api` directory, then you would need to add some additional [configuration in a vercel.json](https://vercel.com/docs/projects/project-configuration).

We have a guide covering [how to use Express.js with Vercel](https://vercel.com/guides/using-express-with-vercel) that can help you get it configured. And our Functions documentation has some additional info: [vercel.com/docs/functions](https://vercel.com/docs/functions)

**Reply by jcooler** [reply]
I added an index.ts to an api file and imported app.ts to it which has my endpoints, and my testing endpoint works, but it doesn't seem any other endpoint that is pointing at MongoDB is working. Is there additional setup for mongo on vercel's or mongo's side? Seems the requests are timing out when trying with postman:

every other endpoint besides test endpoint:

An error occurred with your deployment

FUNCTION_INVOCATION_TIMEOUT

test endpoint:
Test endpoint is working

**Reply by jcooler** [reply]
I was able to get it all set up, I had many issues to sort out as I made a ton of assumptions. I was able to find the documentation on Mongo integration with Vercel. Wonderful product, thank you for the help!
